### PR TITLE
Allows you to use the abandoned white ship as an escape shuttle

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
And adds support for designating any ship as escape worthy.

I have enabled the white ship for use as an escape route on Box, Meta, and Dream.

:cl: Kor
rscadd: You can now use the abandoned white ship as an escape route at round end. This is possible on Box, Meta, and Dream.
rscadd: Added support so that mappers can make any shuttle function as an escape shuttle.
/:cl:

### CODE REVIEW

**Comment by Zilenan91** [comment]
How does this work exactly? Does the white ship take off to centcom if it's in Deep Space at the end of a round?

**Comment by neersighted** [comment]
Looks good! Code is perfect too

**Comment by KorPhaeron** [comment]
@Zilenan91 

Takes off from anywhere when the escape shuttle does

**Comment by Zilenan91** [comment]
How would it taking off while it was in transit work?

**Comment by KorPhaeron** [comment]
It'd just go to transit again I guess?

**Comment by PKPenguin321** [comment]
Where does it dock at centcomm

**Comment by KorPhaeron** [comment]
@PKPenguin321 

>//Mapping a new docking point for each ship mappers could potentially want docking with centcomm would take up lots of space, just let them keep flying off into the sunset for their greentext

**Comment by PKPenguin321** [comment]
It should dock on top of emergency shuttle and gib everybody when it does if you emag it

File: code/controllers/subsystem/shuttles/emergency.dm [hunk 1]
```diff
@@ -118,24 +114,30 @@
 				mode = SHUTTLE_STRANDED
 
 			if(time_left <= 0 && !SSshuttle.emergencyNoEscape)
-				//move each escape pod to its corresponding transit dock
-				for(var/obj/docking_port/mobile/pod/M in SSshuttle.mobile)
-					if(M.launch_status == UNLAUNCHED) //Will not launch from the mine/planet
+				//move each escape pod (or applicable spaceship) to its corresponding transit dock
+				for(var/obj/docking_port/mobile/M in SSshuttle.mobile)
```
Comment (RemieRichards) [inline code/controllers/subsystem/shuttles/emergency.dm]: You know what I'm going to say.

File: code/controllers/subsystem/shuttles/emergency.dm [hunk 2]
```diff
@@ -118,24 +114,30 @@
...
 				enterTransit()
 				mode = SHUTTLE_ESCAPE
+				launch_status = ENDGAME_LAUNCHED
 				timer = world.time
 				priority_announce("The Emergency Shuttle has left the station. Estimate [timeLeft(600)] minutes until the shuttle docks at Central Command.", null, null, "Priority")
 		if(SHUTTLE_ESCAPE)
 			if(time_left <= 0)
 				//move each escape pod to its corresponding escape dock
-				for(var/obj/docking_port/mobile/pod/M in SSshuttle.mobile)
+				for(var/obj/docking_port/mobile/M in SSshuttle.mobile)
```
Comment (RemieRichards) [inline code/controllers/subsystem/shuttles/emergency.dm]: ditto

File: code/game/atoms.dm [hunk 3]
```diff
@@ -27,9 +27,8 @@
 		return 0
 
 	//check for centcomm shuttles
-	for(var/centcom_shuttle in list("emergency", "pod1", "pod2", "pod3", "pod4", "ferry"))
-		var/obj/docking_port/mobile/M = SSshuttle.getShuttle(centcom_shuttle)
-		if(T in M.areaInstance)
+	for(var/obj/docking_port/mobile/M in SSshuttle.mobile)
```
Comment (RemieRichards) [inline code/game/atoms.dm]: ditto~

**Comment by Dorsisdwarf** [comment]
Make it ram the Pod Recovery Ship

**Comment by Iamgoofball** [comment]
just ram right into CC tbh

**Comment by tkdrg** [comment]
Wow kor got it right from the first try, nice job

**Comment by KorPhaeron** [comment]
@tkdrg 

No bully

**Pull request merged**